src/meta/src/stream/source_manager.rs

```diff
@@ -588,6 +588,32 @@ where
     )
 }
 
+pub fn validate_assignment(assignment: &mut HashMap<ActorId, Vec<SplitImpl>>) {
+    // check if one split is assign to multiple actors
+    let mut split_to_actor = HashMap::new();
+    for (actor_id, splits) in &mut *assignment {
+        let _ = splits.iter().map(|split| {
+            split_to_actor
+                .entry(split.id())
+                .or_insert_with(Vec::new)
+                .push(*actor_id)
+        });
+    }
+
+    for (split_id, actor_ids) in &mut split_to_actor {
+        if actor_ids.len() > 1 {
+            tracing::warn!(split_id = ?split_id, actor_ids = ?actor_ids, "split is assigned to multiple actors");
```

Comment:
I tend to panic here to expose the problem early. Otherwise, it will almost certainly trigger a panic in storage side, right?

Reply:
We don't know the root cause and have no way to fix the persisted data. From the frequency the users meet the problem, letting it panic and telling users to drop all sources can hurt the users' experience.

Reply:
> Otherwise, it will almost certainly trigger a panic in storage side, right?

Oh, the following code will fix the multiple assignment issue and the compactor will no longer panic.

Reply:
> > Otherwise, it will almost certainly trigger a panic in storage side, right?
> 
> Oh, the following code will fix the multiple assignment issue and the compactor will no longer panic.

I see your point